src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -1,52 +1,77 @@
     }
 
     public PineconeIndexOperationClient(PineconeClientConfig clientConfig) {
-        this.clientConfig = clientConfig;
-        client = new DefaultAsyncHttpClient();
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this(clientConfig, new OkHttpClient());
+    }
+
+    private static String createUrl(PineconeClientConfig clientConfig) {
+        if (clientConfig.getApiKey() == null || clientConfig.getEnvironment() == null) {
+            throw new PineconeValidationException("Both API key and environment name are required for index operations.");
+        }
+
+        return BASE_URL_PREFIX + clientConfig.getEnvironment() + BASE_URL_SUFFIX;
     }
 
     public void deleteIndex(String indexName) throws IOException {
-        System.out.println("Sending delete index request:");
-        client.prepare("DELETE", url + indexName)
-                .setHeader("accept", "text/plain")
-                .setHeader("Api-Key", clientConfig.getApiKey())
-                .execute()
-                .toCompletableFuture()
-                .join();
-
-        client.close();
+        Request request = new Request.Builder()
+                .url(url + indexName)
+                .delete()
+                .addHeader(ACCEPT_HEADER, TEXT_PLAIN)
+                .addHeader(API_KEY, clientConfig.getApiKey())
+                .build();
+
+        try (Response response = client.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw new IOException(response.message());
+            }
+        } finally {
+            close(client);
+        }
     }
 
     public void createIndex(CreateIndexRequest createIndexRequest) throws IOException {
-        client.prepare("POST", url)
-                .setHeader("accept", "text/plain")
-                .setHeader("content-type", "application/json")
-                .setHeader("Api-Key", clientConfig.getApiKey())
-                .setBody(createIndexRequest.toJson())
-                .execute()
-                .toCompletableFuture()
-                .join();
-
-        client.close();
-    }
-}
+        MediaType mediaType = MediaType.parse("application/json; charset=utf-8");
+        RequestBody requestBody = RequestBody.create(createIndexRequest.toJson(), mediaType);
 
+        Request request = new Request.Builder()
+                .url(url)
+                .post(requestBody)
+                .addHeader(ACCEPT_HEADER, TEXT_PLAIN)
+                .addHeader(CONTENT_TYPE, CONTENT_TYPE_JSON)
+                .addHeader(API_KEY, clientConfig.getApiKey())
+                .build();
 
+        try (Response response = client.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw new IOException(response.message());
+            }
+        } finally {
+            close(client);
+        }
+    }
+
+    public void close(OkHttpClient client) {
```

Comment:
From the OkHttpClient docs

> Shutdown the dispatcher’s executor service with [shutdown()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#shutdown--). This will also cause future calls to the client to be rejected.

This is another reason we shouldn't automatically call close(). I haven't done hands-on testing, but my reading of this makes me think the way we are calling close after every operation will render the http client unable to be used again. So you won't be able to do two or more operations in a row without creating a new instance of `PineconeIndexOperationClient` (to get a new okHttpClient instance) which is definitely not what we want.

Reply:
Yes, that's the reason why I didn't like it in the first place but this will be removed.